boot/initramfs.go

```diff
@@ -125,3 +127,65 @@ func MockInitramfsReboot(f func() error) (restore func()) {
 func InitramfsReboot() error {
 	return initramfsReboot()
 }
+
+// This function implements logic that is usually part of the
+// bootloader, but that it is not possible to implement in, for
+// instance, piboot. See handling of kernel_status in
+// bootloader/assets/data/grub.cfg.
+func updateNotScriptableBootloaderStatus(bl bootloader.NotScriptableBootloader) error {
+	blVars, err := bl.GetBootVars("kernel_status")
+	if err != nil {
+		return err
+	}
+	curKernStatus := blVars["kernel_status"]
+	if curKernStatus == "" {
+		return nil
+	}
+
+	cmdLine, err := osutil.KernelCommandLine()
+	if err != nil {
+		return err
+	}
+	args := strings.Split(cmdLine, " ")
+	// "" would be the value for the error case, which at this point is any
+	// case different to kernel_status=trying in kernel command line and
+	// kernel_status=try in configuration file. Note that kernel_status in
+	// the file should be only "try" or empty, and for the latter we should
+	// have returned a few lines up.
+	newStatus := ""
+	for _, arg := range args {
+		if arg == "kernel_status=trying" && curKernStatus == "try" {
+			newStatus = "trying"
+			break
+		}
+	}
```

Comment:
couldn't we use `KernelCommandLineKeyValues` for this instead of comparing to `kernel_status=trying` directly?

Reply:
Done now